pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go

```diff
@@ -225,6 +240,48 @@ func (mutator *VMsMutator) setDefaultPreferenceKind(vm *v1.VirtualMachine) {
 	}
 }
 
+func validateInstancetypeMatcherUpdate(oldInstancetypeMatcher *v1.InstancetypeMatcher, newInstancetypeMatcher *v1.InstancetypeMatcher) []metav1.StatusCause {
+	if oldInstancetypeMatcher == nil && newInstancetypeMatcher == nil {
+		return nil
+	}
+	if err := validateMatcherUpdate(*oldInstancetypeMatcher, *newInstancetypeMatcher); err != nil {
+		return []metav1.StatusCause{{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: err.Error(),
+			Field:   k8sfield.NewPath("spec", "instancetype", "revisionName").String(),
+		}}
+	}
+	return nil
+}
+
+func validatePreferenceMatcherUpdate(oldPreferenceMatcher *v1.PreferenceMatcher, newPreferenceMatcher *v1.PreferenceMatcher) []metav1.StatusCause {
+	if oldPreferenceMatcher == nil && newPreferenceMatcher == nil {
+		return nil
+	}
+	if err := validateMatcherUpdate(*oldPreferenceMatcher, *newPreferenceMatcher); err != nil {
+		return []metav1.StatusCause{{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: err.Error(),
+			Field:   k8sfield.NewPath("spec", "preference", "revisionName").String(),
+		}}
+	}
+	return nil
+}
+
+func validateMatcherUpdate[M v1.Matcher](oldMatcher M, newMatcher M) error {
```

Comment:
Why generic?

Reply:
@xpivarc hopefully that answered your question, would you mind taking another look?